dPhysics/dpEntity.h

```diff
@@ -80,7 +80,7 @@ class dpEntity {
 
 	bool m_IsGargantuan = false;
 
-	std::vector<dpEntity*> m_NewObjects;
-	std::vector<dpEntity*> m_RemovedObjects;
-	std::map<LWOOBJID, dpEntity*> m_CurrentlyCollidingObjects;
+	std::vector<LWOOBJID> m_NewObjects;
+	std::vector<LWOOBJID> m_RemovedObjects;
+	std::vector<LWOOBJID> m_CurrentlyCollidingObjects;
```

Comment:
This should be a set for O(1)* time.

Reply:
a set is actually slower here.  The only reason I would support a set is to prevent double registration of currently colliding objects.

Reply:
This set is generally very small (really only players are colliding with others, and there are only a few doing so) and because of this, the locality of reference overpowers the lookup speed for a set (unordered_set is the O(1) set which I think you meant to suggest, set in c++ is ordered and O(log base2 n)) in the grand scheme of things we wont really notice a difference, but with small containers under around 6-7 elements, a vector search in linear time is faster than a tree search in logn time.  for an unordered set however, they are roughly equal depending on the under the hood implementation of the items (the actual elements may be stored in a vector and provided via an unordered interface).

Reply:
in our context given what a LWOOBJID is and how msvc/gcc/clang implement the hash for a unsigned long long, an unordered_set should be the fastest option here, as you suggested Wincent.  the average number of elements in a collision can be high in populated worlds, so a set is correct here.

Reply:
Yeah apologizes for the ambiguity. I almost always resort to unordered_set when I talk about sets, almost always what you want.

Reply:
the naming of sets should have been reversed (default set unordered and an ordered_set) but alas a decision made 45 years ago echos to this day :)

Reply:
Are we actually sure we want it to be unordered too? We are doing a lot of searches, insertions, and removals, aren’t we? Using an ordered set penalizes the insertions and removals but it makes the searches faster.

Reply:
Also, I am somewhat confused about the benefits of an unordered_set vs the vector approach I was using initially. I was using a "swap and pop" O(1) deletion scheme as well as appending new elements to the back. While finding an element required iterating until a match was found, that iteration was also occurring over continuous memory. Is an unordered_set really more advantageous than that?

Reply:
They are about equal given the hash scheme for a lwoobjid.  When taking into account social worlds, they may have a size of 20+, which is well beyond where a vector locality overpowers the lookup speed